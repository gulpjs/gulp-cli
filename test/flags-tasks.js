'use strict';

var lab = exports.lab = require('lab').script();
var expect = require('code').expect;
var fs = require('fs');
var path = require('path');
var skipLines = require('./tools/skip-lines');
var eraseTime = require('./tools/erase-time');
var runner = require('./tools/run-gulp');

var expectedDir = path.join(__dirname, 'expected');

lab.experiment('flag: --tasks', function() {

  lab.test('prints the task list', function(done) {
    runner({ verbose: false })
      .gulp('--tasks --cwd ./fixtures/gulpfiles')
      .run(cb);

    function cb(err, stdout) {
      var filepath = path.join(expectedDir, 'flags-tasks.txt');
      var expected = fs.readFileSync(filepath, 'utf-8');
      stdout = eraseTime(skipLines(stdout, 1));
      expect(stdout).to.equal(expected);
      done();
    }
  });

  lab.test('print the task list with description and flags', function(done) {
    runner({ verbose: false })
      .gulp('--tasks',
        '--gulpfile ./fixtures/gulpfiles/with-desc-and-flags.js',
        '--cwd ./fixtures')
      .run(cb);

    function cb(err, stdout) {
      var filepath = path.join(expectedDir, 'with-desc-and-flags.txt');
      var expected = fs.readFileSync(filepath, 'utf-8');
      stdout = eraseTime(skipLines(stdout, 1));
      expect(stdout).to.equal(expected);
      done();
    }
  });

  lab.test('print the task list by gulp.task(s).unwrap and gulp.task(s)',
  function(done) {
    runner({ verbose: false })
      .gulp('--tasks',
        '--gulpfile ./fixtures/gulpfiles/by-unwrap-and-not-by-unwrap.js',
        '--cwd ./fixtures')
      .run(cb);

    function cb(err, stdout) {
      var filepath = path.join(expectedDir, 'by-unwrap-and-not-by-unwrap.txt');
      var expected = fs.readFileSync(filepath, 'utf-8');
      stdout = eraseTime(skipLines(stdout, 1));
      expect(stdout).to.equal(expected);
      done();
    }
  });

});
