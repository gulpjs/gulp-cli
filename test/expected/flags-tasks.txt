gulp-cli/test/fixtures/gulpfiles
├─┬ default
│ └─┬ <series>
│   ├─┬ test1
│   │ └─┬ <series>
│   │   └── noop
│   ├─┬ test3
│   │ └─┬ <series>
│   │   └── described
│   └── noop
├─┬ test1
│ └─┬ <series>
│   └── noop
├─┬ test2
│ └─┬ <series>
│   ├─┬ test1
│   │ └─┬ <series>
│   │   └── noop
│   └── noop
├─┬ test3
│ └─┬ <series>
│   └── described
└─┬ test4
  └─┬ <series>
    ├── errorFunction
    └── anon
